clar2wasm/src/wasm_generator.rs

```diff
@@ -1028,7 +1074,12 @@ impl WasmGenerator {
 
         // Traverse the last statement in the block, whose result is the result
         // of the `begin` expression.
-        self.traverse_expr(builder, statements.last().unwrap())
+        self.traverse_expr(
+            builder,
+            statements.last().ok_or_else(|| {
```

Comment:
There is an assertion above that ensures that the list is not empty, so I think the `unwrap` is fine here. Though we should change the `assert!` above to return an error instead or asserting.

Reply:
Fixed